rfc/0022-policy-lifecycle.md

```diff
@@ -0,0 +1,166 @@
+
+The proposed changes aim to address these issues by introducing a new policy lifecycle that includes the following features:
+
+- Hot reload of policies: The Policy Server will be able to load a new policy without requiring a restart.
+- Policy validation: The Policy Server will validate the policy before loading it, ensuring that it is correctly compiled and can be loaded.
+- Policy status monitoring: The Policy Server will report the status of the policy, indicating whether it was loaded successfully or if an error occurred.
+- Policy versioning: The Policy Server will keep running with a previous version of the policy in case of issues with the latest policy.
+- Policy rollback: The user will be able to rollback to a previous policy version in case of issues with the latest policy.
+
+## Examples / User Stories
+
+[examples]: #examples
+
+- As a user, I want to load a new policy/update an existing policy without requiring a restart of the policy server.
+- As a user, I want to know if a policy was pulled or validated successfully.
+- As a user, I want to monitor the status of a policy on specific policy server replica so that I can identify and resolve issues with the policy.
+- As a user, I want that the policy server keeps running with a previous version of the policy in case of issues with the latest policy.
+- As a user, I want to rollback to a previous policy version in case of issues with the latest policy.
+
+# Detailed design
+
+[design]: #detailed-design
+
+## Concepts
+
+To implemenet the new policy lifecycle, we will introduce the following concepts:
+
+### The Policy Server acts as a controller
+
+At the time of writing the Policy Server is configured with a ConfigMap that contains all the policies to be loaded.
+When a new policy is created, the ConfigMap is updated, and the Policy Server is restarted to load the new policy.
+
+We want to change this behavior to allow the Policy Server to act as a controller that can load new policies without requiring a restart.
+The Policy Server will be watching the Policy CRD directly and it will change the internal state of the evaluation environment when a new policy is created or updated.
+
+A leader election mechanism will be implemented to ensure that only one Policy Server instance is acting as a controller at a time.
+The leader will be responsible to pull the policies from the registry, validate the policy settings, precompile the policy and store it in a shared cache.
+After the policy is precompiled, the leader will change the status of the policy to Valid and signal the other replicas that a new policy is available to be loaded.
+The replicas will be watching the Policy CRDs and load the new policy when the status is `Valid`.
+In case of an error, the policy status will be updated, but the Policy Server will continue serving the previous valid policy until the issue is resolved.
+
+### Shared Policy Cache
```

Comment:
what is a cache? mention wasmtime precompiled modules and backward compatibility.